syntax = "proto3";

package inventoryService;

import "book.proto";

enum CreateBookStatus {
    UNDEFINED = 0;
    SUCCESS = 1;
    ISBN_ERROR = 2;
    TITLE_ERROR = 3;
    AUTHOR_ERROR = 4;
    GENRE_ERROR = 5;
    YEAR_ERROR = 6;
}

service InventoryService{
//    a. CreateBook - self-explanatory
//    Hint: think of a response format that doesnâ€™t include an object
    rpc CreateBook(CreateBookRequest) returns (CreateBookResponse) {}

//    b. GetBook - given ISBN, retrieve book details
    rpc GetBook(GetBookRequest) returns (CreateBookResponse) {}

}

message CreateBookRequest{
    books.Book book = 1;
}

message CreateBookResponse{
    CreateBookStatus status = 1;
    string message = 2;
}

message GetBookRequest{
    string isbn = 1;
}

message GetBookResponse{
    books.Book book = 1;
}